# Introduction to Probability

## Step 1 – Understand the Concept

**Probability** is a measure of how likely an event is to happen.

The probability value is always between **0 and 1**:

- **0** → The event is impossible.
- **1** → The event is certain.
- A value between 0 and 1 → The event has some chance of happening.

### Basic Formula

Probability of an event:

P(Event) = Number of favorable outcomes / Total number of possible outcomes

### Simple Example

When we toss a fair coin:

- Possible outcomes = Head, Tail
- Favorable outcomes for getting Head = 1
- Total outcomes = 2

Therefore:

P(Head) = 1 / 2 = 0.5

So, there is a **50% chance** of getting Head.

## Step 2 – Demonstrate the Concept

Imagine rolling a fair six-sided die.

Possible outcomes:

1, 2, 3, 4, 5, 6

Suppose we want to find the probability of getting an even number.

Even numbers are:

2, 4, 6

So:

- Favorable outcomes = 3
- Total outcomes = 6

Probability = 3 / 6 = 0.5

Therefore, the probability of getting an even number is **0.5 or 50%**.

### Real-World Examples

Probability is used to estimate:

- Whether a customer will purchase a product
- Whether an email is spam
- Whether a machine will fail
- Whether a loan application may be approved
- The likelihood of a disease based on test results

In [1]:
# Define the total number of possible outcomes
total_outcomes = 6

# Define the number of favorable outcomes
favorable_outcomes = 3

# Calculate probability
probability = favorable_outcomes / total_outcomes

# Display the result
print("Probability of getting an even number:", probability)
print("Probability as a percentage:", probability * 100, "%")

Probability of getting an even number: 0.5
Probability as a percentage: 50.0 %


## Step 3 – Explanation

### What does the program do?

The program calculates the probability of getting an even number when rolling a six-sided die.

There are 6 possible outcomes and 3 favorable outcomes: 2, 4, and 6.

### What statistical concept does it demonstrate?

It demonstrates the basic concept of **Probability**.

The probability is calculated by dividing the number of favorable outcomes by the total number of possible outcomes.

### How do we interpret the output?

The output will be:

Probability = 0.5

Percentage = 50%

This means there is a **50% chance** of getting an even number when rolling a fair die.

### Where is this used in AI/ML?

Probability is a fundamental concept in AI and Machine Learning.

It is used in:

- Classification
- Prediction
- Spam detection
- Recommendation systems
- Risk analysis
- Bayesian models
- Uncertainty estimation

Machine learning models often use probability to represent how likely a particular outcome or prediction is.

# Events

## Step 1 – Understand the Concept

In probability, an **event** is a specific outcome or a group of outcomes that we are interested in.

An event is usually represented by a letter such as **A, B, or C**.

### Simple Example

When rolling a six-sided die:

Possible outcomes:

`1, 2, 3, 4, 5, 6`

Suppose Event A is:

**Getting an even number**

Then:

`A = {2, 4, 6}`

So, Event A contains all outcomes that satisfy our condition.

### Types of Events

- **Simple Event** – Contains one outcome.
- **Compound Event** – Contains more than one outcome.
- **Certain Event** – An event that will definitely happen.
- **Impossible Event** – An event that cannot happen.

## Step 2 – Demonstrate the Concept

Consider rolling a six-sided die.

### Event A – Getting an even number

`A = {2, 4, 6}`

There are 3 favorable outcomes.

### Event B – Getting a number greater than 4

`B = {5, 6}`

There are 2 favorable outcomes.

### Real-World Example

Suppose an ML model predicts whether an email is spam.

The event can be:

**Event A = Email is classified as spam**

The possible outcomes could be:

- Spam
- Not Spam

The event represents the particular outcome we are interested in.

In [2]:
# Define the possible outcomes of rolling a die
outcomes = {1, 2, 3, 4, 5, 6}

# Define an event: getting an even number
event_even = {number for number in outcomes if number % 2 == 0}

# Display the event
print("Sample outcomes:", outcomes)
print("Event - Even numbers:", event_even)

# Calculate the probability of the event
probability = len(event_even) / len(outcomes)

print("Probability of the event:", probability)

Sample outcomes: {1, 2, 3, 4, 5, 6}
Event - Even numbers: {2, 4, 6}
Probability of the event: 0.5


## Step 3 – Explanation

### What does the program do?

The program creates the possible outcomes of rolling a six-sided die.

It then identifies the outcomes that belong to the event of getting an even number.

Finally, it calculates the probability of that event.

### What statistical concept does it demonstrate?

It demonstrates the concept of an **Event in Probability**.

The event contains the outcomes that satisfy a particular condition.

Here:

`Event = {2, 4, 6}`

### How do we interpret the output?

The program identifies 3 favorable outcomes out of 6 possible outcomes.

Therefore:

`Probability = 3 / 6 = 0.5`

So, the probability of the event is **50%**.

### Where is this used in AI/ML?

Events are useful when defining outcomes or conditions in probabilistic models.

For example:

- Email is spam
- Customer will purchase
- Machine will fail
- Image contains a particular object
- Model prediction belongs to a particular class

Defining events helps us calculate and reason about probabilities in AI/ML.

# Sample Space

## Step 1 – Understand the Concept

The **Sample Space** is the set of **all possible outcomes** of a probability experiment.

It is usually represented by the symbol **S**.

### Simple Example

When tossing a coin, the possible outcomes are:

`Head, Tail`

Therefore:

`S = {Head, Tail}`

When rolling a six-sided die:

`S = {1, 2, 3, 4, 5, 6}`

The sample space contains every possible outcome that can occur.

### Important Point

An **event** is a subset of the sample space.

For example:

`S = {1, 2, 3, 4, 5, 6}`

Event A = getting an even number:

`A = {2, 4, 6}`

So, `A` is part of the sample space `S`.

## Step 2 – Demonstrate the Concept

Consider tossing a coin twice.

The possible outcomes are:

- Head, Head → HH
- Head, Tail → HT
- Tail, Head → TH
- Tail, Tail → TT

Therefore, the sample space is:

`S = {HH, HT, TH, TT}`

There are **4 possible outcomes**.

### Real-World Example

In an AI/ML classification problem, suppose a model predicts whether an image contains:

- Cat
- Dog
- Bird

The possible prediction outcomes form the sample space:

`S = {Cat, Dog, Bird}`

The sample space helps us understand all possible outcomes before calculating probabilities.

In [3]:
# Define the possible outcomes of tossing a coin twice
sample_space = {"HH", "HT", "TH", "TT"}

# Display the sample space
print("Sample Space:", sample_space)

# Count the total number of possible outcomes
total_outcomes = len(sample_space)

print("Total number of outcomes:", total_outcomes)

Sample Space: {'HH', 'TT', 'HT', 'TH'}
Total number of outcomes: 4


## Step 3 – Explanation

### What does the program do?

The program creates the sample space for tossing a coin twice.

It stores all four possible outcomes:

`HH, HT, TH, TT`

It then counts the total number of possible outcomes.

### What statistical concept does it demonstrate?

It demonstrates the concept of a **Sample Space**.

The sample space contains every possible outcome of a probability experiment.

### How do we interpret the output?

The output contains:

`HH, HT, TH, TT`

The total number of possible outcomes is:

**4**

This means there are four different outcomes when a coin is tossed twice.

### Where is this used in AI/ML?

Sample spaces help define all possible outcomes in probabilistic problems.

They can be useful in:

- Classification problems
- Prediction
- Risk analysis
- Decision-making
- Probabilistic models
- Bayesian reasoning

Defining the possible outcomes is an important first step before calculating probabilities.

# Independent Events

## Step 1 – Understand the Concept

**Independent Events** are events where the occurrence of one event does **not affect** the occurrence of another event.

The probability of one event remains the same even if we know that the other event has occurred.

### Formula

For independent events A and B:

P(A and B) = P(A) × P(B)

### Simple Example

Suppose we toss a coin twice.

- Event A = First toss is Head
- Event B = Second toss is Head

The first toss does not affect the second toss.

Therefore, these are **independent events**.

P(A) = 1/2

P(B) = 1/2

So:

P(A and B) = 1/2 × 1/2 = 1/4

## Step 2 – Demonstrate the Concept

Imagine tossing a fair coin twice.

### Event A

First toss = Head

Probability:

`P(A) = 1/2`

### Event B

Second toss = Head

Probability:

`P(B) = 1/2`

The result of the first toss does not change the result of the second toss.

Therefore:

`P(A and B) = P(A) × P(B)`

`= 1/2 × 1/2`

`= 1/4`

So, there is a **25% probability** of getting Head on both tosses.

### Real-World Example

In AI/ML, independent features or events can sometimes be assumed to simplify probabilistic models.

For example, some Naive Bayes models assume that features are conditionally independent given the class.

In [4]:
# Probability of getting Head on the first toss
probability_A = 1 / 2

# Probability of getting Head on the second toss
probability_B = 1 / 2

# Calculate the probability of both events occurring
probability_both = probability_A * probability_B

# Display the result
print("Probability of Head on first toss:", probability_A)
print("Probability of Head on second toss:", probability_B)
print("Probability of Head on both tosses:", probability_both)
print("Percentage:", probability_both * 100, "%")

Probability of Head on first toss: 0.5
Probability of Head on second toss: 0.5
Probability of Head on both tosses: 0.25
Percentage: 25.0 %


## Step 3 – Explanation

### What does the program do?

The program calculates the probability of getting Head on both tosses of a fair coin.

It multiplies the probabilities of the two independent events.

### What statistical concept does it demonstrate?

It demonstrates **Independent Events**.

When two events are independent, the probability of both events occurring is calculated by multiplying their individual probabilities.

### How do we interpret the output?

The output will be:

`Probability = 0.25`

`Percentage = 25%`

This means there is a **25% chance** of getting Head on both coin tosses.

### Where is this used in AI/ML?

Independence assumptions are used in several probabilistic AI/ML methods.

A common example is **Naive Bayes**, where features are assumed to be conditionally independent given the class.

Understanding independent events helps us understand how probabilities can be combined in probabilistic models.

# Dependent Events

## Step 1 – Understand the Concept

**Dependent Events** are events where the occurrence of one event **affects the probability of another event**.

In other words, the outcome of the first event changes the possible outcomes or probability of the second event.

### Formula

For dependent events A and B:

P(A and B) = P(A) × P(B | A)

Here:

- P(A) = Probability of event A
- P(B | A) = Probability of event B occurring given that A has already occurred

### Simple Example

Suppose a bag contains:

- 3 red balls
- 2 blue balls

We pick one ball **without replacing it**, and then pick another ball.

If the first ball is red, the number of red balls remaining changes.

Therefore, the probability of the second pick depends on the first pick.

These are **Dependent Events**.

## Step 2 – Demonstrate the Concept

Suppose a bag contains:

`3 Red + 2 Blue = 5 balls`

We pick two balls without replacement.

### Event A

First ball is Red.

`P(A) = 3/5`

After picking a red ball:

- Red balls remaining = 2
- Total balls remaining = 4

### Event B

Second ball is Red given that the first ball was Red.

`P(B | A) = 2/4 = 1/2`

Therefore:

`P(A and B) = 3/5 × 2/4`

`= 3/10`

So, the probability of picking two red balls is **30%**.

### Real-World Example

Dependent events can occur in AI/ML when the probability of one observation or outcome changes based on another observation.

They are important in:

- Bayesian models
- Sequential predictions
- Time-series data
- Medical diagnosis
- Risk analysis

In [5]:
# Total number of balls
total_balls = 5

# Number of red balls
red_balls = 3

# Probability of picking a red ball first
probability_first_red = red_balls / total_balls

# After picking one red ball
remaining_red_balls = 2
remaining_balls = 4

# Probability of picking a red ball second
probability_second_red = remaining_red_balls / remaining_balls

# Probability of picking two red balls
probability_two_red = probability_first_red * probability_second_red

# Display the results
print("Probability of first red:", probability_first_red)
print("Probability of second red given first red:", probability_second_red)
print("Probability of two red balls:", probability_two_red)
print("Percentage:", probability_two_red * 100, "%")

Probability of first red: 0.6
Probability of second red given first red: 0.5
Probability of two red balls: 0.3
Percentage: 30.0 %


## Step 3 – Explanation

### What does the program do?

The program calculates the probability of picking two red balls from a bag without replacing the first ball.

It calculates the first probability and then updates the probability for the second pick.

### What statistical concept does it demonstrate?

It demonstrates **Dependent Events**.

The second probability changes because the first ball was removed from the bag.

### How do we interpret the output?

The output will show:

- First red probability = 0.6
- Second red probability given first red = 0.5
- Probability of two red balls = 0.3

Therefore, there is a **30% chance** of picking two red balls.

### Where is this used in AI/ML?

Dependent events are important when the outcome of one observation affects another.

They are used in:

- Bayesian inference
- Sequential prediction
- Time-series analysis
- Medical diagnosis
- Probabilistic models

Understanding dependency helps AI/ML models represent relationships between observations.

# Conditional Probability

## Step 1 – Understand the Concept

**Conditional Probability** is the probability of an event occurring **given that another event has already occurred**.

It helps us calculate probability when we have additional information.

### Formula

P(A | B) = P(A and B) / P(B)

Here:

- P(A | B) = Probability of A given B
- P(A and B) = Probability that both A and B occur
- P(B) = Probability of event B

### Simple Example

Suppose a class has 20 students:

- 12 students know Python.
- 8 of those 12 students also know SQL.

If we already know that a student knows Python, we can calculate the probability that the student also knows SQL.

P(SQL | Python) = 8 / 12

= 0.667

So, the probability is approximately **66.7%**.

## Step 2 – Demonstrate the Concept

Consider a dataset of 100 customers.

- 40 customers purchased a product.
- 30 customers who purchased the product also used a discount coupon.

We want to find:

**Probability that a customer used a coupon given that they purchased the product.**

P(Coupon | Purchased)

= Customers who purchased and used coupon / Total customers who purchased

= 30 / 40

= 0.75

Therefore, the conditional probability is **75%**.

The important point is that we are calculating the probability **within the group of customers who purchased the product**.

### Real-World Example

In AI/ML, conditional probability is used when making predictions based on known information.

For example:

**What is the probability that an email is spam given that it contains certain words?**

In [6]:
# Number of customers who purchased the product
purchased = 40

# Number of customers who purchased and used a coupon
purchased_with_coupon = 30

# Calculate conditional probability
conditional_probability = purchased_with_coupon / purchased

# Display the result
print("P(Coupon | Purchased):", conditional_probability)
print("Percentage:", conditional_probability * 100, "%")

P(Coupon | Purchased): 0.75
Percentage: 75.0 %


## Step 3 – Explanation

### What does the program do?

The program calculates the probability that a customer used a coupon, given that the customer purchased the product.

It considers only the customers who purchased the product.

### What statistical concept does it demonstrate?

It demonstrates **Conditional Probability**.

Conditional probability calculates the likelihood of one event when we already know that another event has occurred.

### How do we interpret the output?

The output will be:

`Probability = 0.75`

`Percentage = 75%`

This means that among the customers who purchased the product, **75% used a coupon**.

### Where is this used in AI/ML?

Conditional probability is widely used in:

- Classification
- Spam detection
- Medical diagnosis
- Recommendation systems
- Bayesian inference
- Natural Language Processing (NLP)

For example, an ML model can estimate the probability of a class given certain features.

# Bayes' Theorem

## Step 1 – Understand the Concept

**Bayes' Theorem** is used to calculate the probability of an event based on **new or additional information**.

It helps us update our belief about an event when we receive new evidence.

### Formula

P(A | B) = [P(B | A) × P(A)] / P(B)

Where:

- **P(A | B)** → Probability of A given B
- **P(B | A)** → Probability of B given A
- **P(A)** → Prior probability of A
- **P(B)** → Probability of the evidence B

### Simple Example

Suppose a medical test is positive.

We want to find:

**What is the probability that a person actually has the disease given that the test is positive?**

Bayes' Theorem helps us calculate this probability using:

- How common the disease is
- How accurate the test is
- How often the test gives a positive result

## Step 2 – Demonstrate the Concept

Consider the following example:

Suppose:

- 1% of people have a particular disease.
- If a person has the disease, the test is positive 90% of the time.
- If a person does not have the disease, the test is positive 5% of the time.

We want to calculate:

**P(Disease | Positive Test)**

Using Bayes' Theorem, we combine the prior probability and the test evidence to calculate the updated probability.

This shows why a positive test result does not automatically mean that the person has the disease.

### Real-World Examples

Bayes' Theorem is used in:

- Medical diagnosis
- Spam email detection
- Fraud detection
- Recommendation systems
- Machine learning classification
- Natural Language Processing

In [7]:
# Prior probability of having the disease
p_disease = 0.01

# Probability of a positive test given that the person has the disease
p_positive_given_disease = 0.90

# Probability of a positive test given that the person does not have the disease
p_positive_given_no_disease = 0.05

# Probability of not having the disease
p_no_disease = 1 - p_disease

# Calculate the total probability of a positive test
p_positive = (
    p_positive_given_disease * p_disease
    + p_positive_given_no_disease * p_no_disease
)

# Apply Bayes' Theorem
p_disease_given_positive = (
    p_positive_given_disease * p_disease
) / p_positive

# Display the result
print("Probability of disease given a positive test:",
      p_disease_given_positive)

print("Percentage:",
      p_disease_given_positive * 100, "%")

Probability of disease given a positive test: 0.15384615384615385
Percentage: 15.384615384615385 %


## Step 3 – Explanation

### What does the program do?

The program calculates the probability that a person actually has the disease given that their test result is positive.

It uses the disease's prior probability and the test's positive-result probabilities.

### What statistical concept does it demonstrate?

It demonstrates **Bayes' Theorem**, which allows us to update the probability of an event using new evidence.

### How do we interpret the output?

The result will be approximately:

**0.1538 or 15.38%**

This means that, based on the given assumptions, the probability of actually having the disease after receiving a positive test is approximately **15.38%**.

This happens because the disease is relatively rare, and some people without the disease can also receive a positive test.

### Where is this used in AI/ML?

Bayes' Theorem is an important concept in probabilistic AI and ML.

It is used in:

- Naive Bayes classification
- Spam detection
- Medical diagnosis
- Fraud detection
- Document classification
- Bayesian inference

It helps models update probabilities when new evidence becomes available.

# Probability Rules

## Step 1 – Understand the Concept

Probability rules are basic mathematical rules used to calculate the probability of events.

### 1. Addition Rule

The addition rule is used to find the probability of **A or B**.

For mutually exclusive events:

P(A or B) = P(A) + P(B)

### 2. Multiplication Rule

The multiplication rule is used to find the probability of **A and B**.

For independent events:

P(A and B) = P(A) × P(B)

### 3. Complement Rule

The complement of an event means the event **does not occur**.

P(Not A) = 1 - P(A)

### Simple Example

For a six-sided die:

- P(1) = 1/6
- P(2) = 1/6

Probability of getting 1 or 2:

P(1 or 2) = 1/6 + 1/6 = 2/6 = 1/3

## Step 2 – Demonstrate the Concept

Consider rolling a six-sided die.

### Addition Rule

What is the probability of getting 1 or 2?

P(1 or 2) = P(1) + P(2)

= 1/6 + 1/6

= 1/3

### Multiplication Rule

Suppose we roll the die twice.

What is the probability of getting 6 both times?

P(6 and 6) = 1/6 × 1/6

= 1/36

### Complement Rule

What is the probability of **not getting 6**?

P(Not 6) = 1 - P(6)

= 1 - 1/6

= 5/6

### Real-World Example

Probability rules are used in AI/ML for:

- Combining possible outcomes
- Calculating prediction probabilities
- Risk analysis
- Classification
- Decision-making

In [8]:
# Probability of getting 1 on a die
p_1 = 1 / 6

# Probability of getting 2 on a die
p_2 = 1 / 6

# Addition Rule: probability of getting 1 or 2
p_1_or_2 = p_1 + p_2

# Probability of getting 6
p_6 = 1 / 6

# Multiplication Rule: probability of getting 6 twice
p_6_and_6 = p_6 * p_6

# Complement Rule: probability of not getting 6
p_not_6 = 1 - p_6

# Display the results
print("P(1 or 2):", p_1_or_2)
print("P(6 and 6):", p_6_and_6)
print("P(Not 6):", p_not_6)

P(1 or 2): 0.3333333333333333
P(6 and 6): 0.027777777777777776
P(Not 6): 0.8333333333333334


## Step 3 – Explanation

### What does the program do?

The program calculates probabilities using three basic probability rules:

- Addition Rule
- Multiplication Rule
- Complement Rule

It uses the outcomes of a six-sided die as an example.

### What statistical concept does it demonstrate?

It demonstrates the basic **Rules of Probability** used to combine or calculate probabilities of events.

### How do we interpret the output?

The results are:

- P(1 or 2) = 1/3 ≈ 0.333
- P(6 and 6) = 1/36 ≈ 0.0278
- P(Not 6) = 5/6 ≈ 0.833

These values represent the likelihood of each event.

### Where is this used in AI/ML?

Probability rules are fundamental to many AI/ML algorithms.

They are used in:

- Classification
- Bayesian models
- Risk prediction
- Decision-making
- Probability-based predictions
- Statistical modelling

Understanding these rules helps us understand how ML models combine probabilities to make predictions.

# Random Variables

## Step 1 – Understand the Concept

A **Random Variable** is a variable that represents the numerical outcome of a random experiment.

It assigns a numerical value to each possible outcome.

There are two main types:

### 1. Discrete Random Variable

It takes **countable values**.

Example:

Number of heads when tossing a coin 3 times:

`0, 1, 2, 3`

### 2. Continuous Random Variable

It can take **any value within a range**.

Example:

- Height
- Weight
- Temperature
- Time

### Simple Example

If we roll a die, the random variable X can represent the number that appears.

`X = {1, 2, 3, 4, 5, 6}`

The value of X depends on the random outcome.

## Step 2 – Demonstrate the Concept

Suppose we toss a coin three times.

Let **X = number of Heads**.

Possible values of X are:

`0, 1, 2, 3`

For example:

- TTT → X = 0
- HTT → X = 1
- HHT → X = 2
- HHH → X = 3

Here, X is a **Discrete Random Variable** because it can take countable numerical values.

### Real-World Example

In AI/ML, a random variable can represent:

- Number of customers
- Number of purchases
- Predicted class
- Temperature
- House price
- Model prediction probability

Random variables help us represent uncertain outcomes mathematically.

In [9]:
import numpy as np

# Simulate 10 coin-toss experiments
# 0 represents Tail and 1 represents Head
coin_tosses = np.random.randint(0, 2, size=(10, 3))

# Calculate the number of Heads in each experiment
number_of_heads = coin_tosses.sum(axis=1)

# Display the results
print("Coin toss results:")
print(coin_tosses)

print("\nNumber of Heads in each experiment:")
print(number_of_heads)

Coin toss results:
[[0 1 1]
 [1 1 1]
 [0 0 1]
 [0 0 0]
 [0 0 0]
 [1 0 1]
 [1 1 0]
 [1 0 1]
 [0 1 0]
 [1 0 1]]

Number of Heads in each experiment:
[2 3 1 0 0 2 2 2 1 2]


## Step 3 – Explanation

### What does the program do?

The program simulates 10 experiments where a coin is tossed 3 times.

It represents:

- 0 → Tail
- 1 → Head

It then counts the number of Heads in each experiment.

### What statistical concept does it demonstrate?

It demonstrates a **Discrete Random Variable**.

The number of Heads is the random variable because its value changes depending on the random coin-toss outcomes.

The possible values are:

`0, 1, 2, 3`

### How do we interpret the output?

Each row represents one experiment.

For example:

`[1, 0, 1]`

means:

- Head
- Tail
- Head

So the random variable X has the value:

`X = 2`

Because there are two Heads.

Since the experiment is random, the values of X can change each time we run the program.

### Where is this used in AI/ML?

Random variables are fundamental to probabilistic machine learning.

They are used to represent:

- Uncertain outcomes
- Features
- Predictions
- Class labels
- Measurements
- Probability distributions

They form the foundation for concepts such as probability distributions, expected value, and statistical modelling.

# Expected Value

## Step 1 – Understand the Concept

**Expected Value** is the average value we would expect from a random experiment if it were repeated many times.

It is a **probability-weighted average** of all possible outcomes.

### Formula

For a discrete random variable:

E(X) = Σ [x × P(X = x)]

Where:

- **x** = possible outcome
- **P(X = x)** = probability of that outcome
- **E(X)** = Expected Value

### Simple Example

Suppose we roll a fair six-sided die.

Possible outcomes:

`1, 2, 3, 4, 5, 6`

Each outcome has a probability of `1/6`.

Expected Value:

E(X) = (1×1/6) + (2×1/6) + ... + (6×1/6)

E(X) = 3.5

So, the expected value of a fair die is **3.5**.

Note: 3.5 cannot appear on a single die roll. It represents the long-run average.

## Step 2 – Demonstrate the Concept

Suppose a game has three possible outcomes:

| Outcome | Probability |
|--------:|------------:|
| ₹100    | 0.2         |
| ₹200    | 0.5         |
| ₹500    | 0.3         |

Expected Value:

E(X) = (100 × 0.2) + (200 × 0.5) + (500 × 0.3)

E(X) = 20 + 100 + 150

E(X) = ₹270

Therefore, the expected value is **₹270**.

This does not mean that we will receive exactly ₹270 in one game. It represents the average outcome we would expect over many repetitions.

### Real-World Examples

Expected value is used in:

- Risk analysis
- Insurance
- Finance
- Business decisions
- Customer behaviour prediction
- Machine learning predictions

In [10]:
# Define possible outcomes
outcomes = [100, 200, 500]

# Define the probability of each outcome
probabilities = [0.2, 0.5, 0.3]

# Calculate the expected value
expected_value = sum(
    outcome * probability
    for outcome, probability in zip(outcomes, probabilities)
)

# Display the result
print("Expected Value: ₹", expected_value)

Expected Value: ₹ 270.0


## Step 3 – Explanation

### What does the program do?

The program calculates the expected value of a game with three possible monetary outcomes.

Each outcome is multiplied by its probability, and the results are added together.

### What statistical concept does it demonstrate?

It demonstrates **Expected Value**, which is the probability-weighted average of possible outcomes.

### How do we interpret the output?

The output will be:

`Expected Value: ₹270`

This means that if the game were repeated many times under the same probabilities, the average outcome would approach approximately **₹270 per game**.

It does not mean every individual game will produce ₹270.

### Where is this used in AI/ML?

Expected value is useful in:

- Risk assessment
- Decision-making
- Reinforcement learning
- Predictive modelling
- Business analytics
- Financial modelling

For example, an AI system can compare possible actions based on their expected outcomes and probabilities.